### PR TITLE
Implement initial Settings screen

### PR DESCRIPTION
**Pull request by dellisd**
Restructured SharedPreferences in dagger graph
Implemented abstraction around shared preferences
Fix height of home fragment recycler view
This PR also bundles in a minor fix for Live Updates. Old updates are now cleared properly and they are now ordered from latest to oldest.

# Checklist
- [x] I ran `./gradlew test connectedAndroidTest detekt`
- [x] I rebased off of `upstream/master`
- [x] I added tests if applicable

### CODE REVIEW

File: app/src/main/java/ca/llamabagel/transpo/trips/data/TripsRepository.kt [hunk 1]
```diff
@@ -84,12 +84,11 @@ class TripsRepository @Inject constructor(
 
     @SuppressLint("ApplySharedPref")
```
Comment (dellisd) [inline app/src/main/java/ca/llamabagel/transpo/trips/data/TripsRepository.kt]: Remove this

File: app/src/main/java/ca/llamabagel/transpo/trips/data/TripsRepository.kt [hunk 2]
```diff
@@ -84,12 +84,11 @@ class TripsRepository @Inject constructor(
...
     suspend fun setGroupByRoute(group: Boolean) = withContext(dispatcherProvider.io) {
```
Comment (dellisd) [inline app/src/main/java/ca/llamabagel/transpo/trips/data/TripsRepository.kt]:
```suggestion
    fun setGroupByRoute(group: Boolean) {
```

File: app/src/main/java/ca/llamabagel/transpo/settings/data/AppSettings.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import io.github.dellisd.quicksave.SettingsProvider
+import io.github.dellisd.quicksave.booleanSetting
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class AppSettings @Inject constructor(provider: SettingsProvider) : SettingsProvider by provider {
+    val groupByRoute by booleanSetting {
+        title = "Group results by Route"
+        caption = "Group trips by their Route number and destination"
```
Comment (GustavoSanMartin) [inline app/src/main/java/ca/llamabagel/transpo/settings/data/AppSettings.kt]: we should add string resources for these

Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/settings/data/AppSettings.kt]: Currently unsupported by the library, but noted (for addition to the library).

Comment (dellisd) [reply app/src/main/java/ca/llamabagel/transpo/settings/data/AppSettings.kt]: I'm not sure I want to inject a `StringProvider` into there when the library could simply handle accepting resource ids directly. Injecting `StringProvider` will be the fall back.

**Pull request merged**